models/task_test.go

```diff
@@ -0,0 +1,80 @@
+package models
+
+import (
+	"reflect"
+	"testing"
+	"time"
+)
+
+func TestTask_Map(t *testing.T) {
+	deadlineInput := "2023-06-01T10:00:00Z"
+	deadline, err := time.Parse(time.RFC3339, deadlineInput)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	createdAt := time.Now()
+	task := Task{
+		ID:          "1",
+		Name:        "Sample Task",
+		CreatedAt:   createdAt,
+		Status:      "Pending",
+		Description: "Task description",
+		Deadline:    deadline,
+		Priority:    2,
+		Assignee:    "Ali",
+	}
+
+	expectedMap := map[string]interface{}{
+		"ID":          "1",
+		"Name":        "Sample Task",
+		"CreatedAt":   createdAt,
+		"Status":      "Pending",
+		"Description": "Task description",
+		"Deadline":    deadline,
+		"Priority":    2,
+		"Assignee":    "Ali",
+	}
+
+	taskMap := task.Map()
+
+	if !reflect.DeepEqual(taskMap, expectedMap) {
+		t.Errorf("Map() returned unexpected map.\nExpected: %+v\nGot: %+v", expectedMap, taskMap)
+	}
+}
```

Comment:
tests should have multiple cases including scenarios like, Task with wrong data, not getting what's expected, etc. and any other edge cases you can think of